src/components/CityWeather/CityWeather.tsx

```diff
@@ -0,0 +1,108 @@
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import LoadingSpinner from "../LoadingSpinner/LoadingSpinner";
+import { API_KEY } from "../Weather/Weather.constant";
+import "./CityWeather.css";
+import { ReactComponent as Loupe } from "../../assets/loupe.svg";
+
+type Main = {
+  temp: number;
+};
+
+type CityWeatherResponse = {
+  main: Main;
+  name: string;
+};
+
+type CityWeatherError = {
+  response: any;
+  message: string;
+};
+
+function CityWeather() {
+  const [data, setData] = useState<CityWeatherResponse | null>(null);
+  const [city, setCity] = useState<string>();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isEmpty, setIsEmpty] = useState<boolean>(true);
+  const [error, setError] = useState<CityWeatherError | null>(null);
+
+  const handleClick = async () => {
+    if (isEmpty) {
+      return;
+    }
+    setIsLoading(true);
+
+    axios
+      .get<CityWeatherResponse>(
+        `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${API_KEY}&units=metric`
+      )
+      .then((response) => {
+        setData(response.data);
+        setCity("");
+        setIsEmpty(true);
+        setIsLoading(false);
+        setError(null);
+      })
+      .catch((error) => {
+        setCity("");
+        setError(error);
+        setIsLoading(false);
+      });
+  };
+
+  useEffect(() => {
+    if (!!city) {
+      setIsEmpty(false);
+    }
+  }, [city]);
+
+  const getErrorView = () => {
```

Comment:
separate component please. its easy to test and easy to read

Reply:
done